### ISSUE TITLE
Editor errors don't match command line outputs

### ISSUE THREAD

**Issue by fedetibaldo** [microsoft/vscode-mypy#174]
Given the following example file
```py
from pydantic import BaseModel


class Test(BaseModel):
    foo: str
    should_error = False


test = Test(foo="bar", should_error=True)
```
The editor gives me the following squiggly line:
![image](https://github.com/microsoft/vscode-mypy/assets/21128334/925607ac-3ab7-4f48-91e4-7e4cc0ff062e)

My workspace settings are the following:
```json5
{
    "[python]": {
        "editor.formatOnSave": true,
        "editor.codeActionsOnSave": {
            "source.organizeImports": true
        },
    },
    "python.analysis.autoImportCompletions": true,
    "python.analysis.indexing": true,
    "python.analysis.packageIndexDepths": [
        {
            "name": "fastapi",
            "depth": 2
        },
    ],
    "python.analysis.typeCheckingMode": "off",

    // See https://github.com/microsoft/vscode-mypy/issues/88#issuecomment-1636167871
    "mypy-type-checker.path": ["${workspaceFolder}/.venv/bin/mypy"],
    "mypy-type-checker.args": ["--config-file", "${workspaceFolder}/pyproject.toml"],

    "isort.importStrategy": "fromEnvironment",
    "isort.args": ["--profile", "black"],
    "black-formatter.importStrategy": "fromEnvironment",
    "flake8.importStrategy": "fromEnvironment",
}
```

And my mypy extension output is:
```
2023-10-03 15:49:59.688 [info] /Users/fedetibaldo/[REDACTED]/.venv/bin/mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-code --no-pretty --config-file /Users/fedetibaldo/[REDACTED]/pyproject.toml --show-error-end /Users/fedetibaldo/[REDACTED]/test.py
2023-10-03 15:49:59.688 [info] CWD Server: /Users/fedetibaldo/[REDACTED]
2023-10-03 15:50:00.051 [info] file:///Users/fedetibaldo/[REDACTED]/test.py :
/Users/fedetibaldo/[REDACTED]/test.py:9:8:9:41: error: Unexpected keyword argument "should_error" for "Test"  [call-arg]
```

However, if I run the same exact command in my CLI, no errors are to be found.

```
❯ /Users/fedetibaldo/[REDACTED]/.venv/bin/mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-code --no-pretty --config-file /Users/fedetibaldo/[REDACTED]/pyproject.toml --show-error-end /Users/fedetibaldo/[REDACTED]/test.py
❯ 
```

This wouldn't be terrible if not for the fact that fixing the in-editor issue will make my terminal command fail!

```py
test = Test(foo="bar", should_error=True)  # type: ignore
```
![image](https://github.com/microsoft/vscode-mypy/assets/21128334/8bc8339f-9563-4e63-bf3c-bd6faa55eb94)
```
❯ /Users/fedetibaldo/[REDACTED]/.venv/bin/mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-code --no-pretty --config-file /Users/fedetibaldo/[REDACTED]/pyproject.toml --show-error-end /Users/fedetibaldo/[REDACTED]/test.py
/Users/fedetibaldo/[REDACTED]/test.py:9: error: Unused "type: ignore" comment
❯ 
```

Why are the two execution environments out of sync?

**Comment by fedetibaldo** [comment]
Sorry about the repeated closure/opening. I had accidentally hit CMD+Enter while writing the description

**Comment by karthiknadig** [comment]
@fedetibaldo Can you provide the python version of the environment? also, do you have any stubs installed? And when you are running it manually are you running it from an activated environment, and was the current working directory same as the one in the logs.

**Comment by fedetibaldo** [comment]
The python version is 3.11.5, confirmed by both the editor and the CLI

![image](https://github.com/microsoft/vscode-mypy/assets/21128334/70880aba-d243-4083-883d-aa8faebeb195)
![image](https://github.com/microsoft/vscode-mypy/assets/21128334/c14dc881-4808-4087-ac6f-53b1da524a91)
```
❯ python --version
Python 3.11.5
❯ which python
/Users/fedetibaldo/[REDACTED]/.venv/bin/python
❯
```
When I run it on my CLI it's through an activated terminal. Sourcing `./.venv/bin/activate` before running the mypy command doesn't change the outcome.

The next 2 outputs aren't too helpful, but hopefully they can demonstrate that the directories are the same (they match)
```
2023-10-05 18:03:53.187 [info] CWD Server: /Users/fedetibaldo/[REDACTED]
```
```
❯ pwd
/Users/fedetibaldo/[REDACTED]
```
If typeshed stubs count, these are the `types-*` packages I've installed. I haven't defined any custom stubs other than them.
```
types-python-dateutil = "^2.8.19.14"
types-psycopg2 = "^2.9.21.14"
types-stripe = "^3.5.2.14"
types-pyjwt = "^1.7.1"
```

Thank you for your time 🙇🏻  I'm available for further questions

**Comment by fedetibaldo** [comment]
I fixed it, apparently.

First, I updated `mypy` from `1.0.1` to `1.5.1`. As soon as I did that, the editor checking still worked but the CLI ran into the same issue described in #145. So, I upgraded to the pre-release version of the extension and set `"mypy-type-checker.preferDaemon": false,`. Now the two outputs (editor and CLI) match, but it's very curios 'cause it's the CLI who is now matching what the editor notified me of all along, not the opposite (as I had expected).

I'm closing the issue but I'm still available for further questions. Thank you
